projects/angular/src/lib/maskito.directive.ts

```diff
@@ -1,52 +1,51 @@
-import {Directive, ElementRef, Inject, Input, OnDestroy} from '@angular/core';
+import {Directive, ElementRef, Inject, Input, OnDestroy, Optional} from '@angular/core';
 import {Maskito, MaskitoOptions} from '@maskito/core';
+import {Observable, Subject} from 'rxjs';
+import {takeUntil} from 'rxjs/operators';
 
-import {MASKITO_OPTIONS} from './maskito-options';
+import {MaskitoOptionsDirective} from './maskito-options.directive';
 
 @Directive({
     selector: 'input[maskito], textarea[maskito]',
 })
 export class MaskitoDirective implements OnDestroy {
+    private readonly destroy$ = new Subject();
     private maskedElement: Maskito | null = null;
 
     constructor(
         @Inject(ElementRef)
         private readonly elementRef: ElementRef<HTMLInputElement | HTMLTextAreaElement>,
-        @Inject(MASKITO_OPTIONS)
-        private readonly defaultOptions: MaskitoOptions,
+        @Optional()
+        @Inject(MaskitoOptionsDirective)
+        private readonly externalOptions$: Observable<MaskitoOptions> | null,
     ) {}
 
     /**
      * When developer has direct access to native input element:
      * ```
      * <input [maskito]="options" />
      * ```
-     * ___
-     * When native input element is hidden somewhere deep inside another component:
-     * ```
-     * @Component({
-     *     template: `
-     *         <your-custom-input>
-     *             <!-- <input maskito /> is somewhere inside -->
-     *         </your-custom-input>
-     *     `,
-     *     providers: [{provide: MASKITO_OPTIONS, useValue: {mask: /^\d+$/}}],
-     * })
-     * export class YourComponent {}
-     * ```
-     * WARNING! This approach is acceptable only for the static mask (no mask option changes).
      */
     @Input('maskito')
     set options(options: MaskitoOptions | '') {
-        this.maskedElement?.destroy();
+        this.destroy$.next();
 
-        this.maskedElement = new Maskito(
-            this.elementRef.nativeElement,
-            options || this.defaultOptions,
-        );
+        if (options) {
+            this.setup(options);
+        } else {
+            this.externalOptions$
+                ?.pipe(takeUntil(this.destroy$))
+                .subscribe(options => this.setup(options));
+        }
     }
 
     ngOnDestroy(): void {
+        this.destroy$.next();
+        this.maskedElement?.destroy();
+    }
+
+    private setup(options: MaskitoOptions): void {
         this.maskedElement?.destroy();
+        this.maskedElement = new Maskito(this.elementRef.nativeElement, options);
```

Comment:
Just a point to keep an eye on, we cannot be sure we have access to native element so in future we would also need some way to get native input without putting explicit `maskito` directive on it, and also make sure we can handle native input dynamic appearance and removal.